src/common/uilocale.cpp

```diff
@@ -743,6 +748,29 @@ int wxUILocale::GetSystemLocale()
 /* static */
 wxVector<wxString> wxUILocale::GetPreferredUILanguages()
 {
+    // The WX_LANGUAGE variable may contain a colon separated list of language
+    // codes in the order of preference. It is modelled after GNU's LANGUAGE:
+    // http://www.gnu.org/software/gettext/manual/html_node/The-LANGUAGE-variable.html
+    wxString languageFromEnv;
+    if (wxGetEnv("WX_LANGUAGE", &languageFromEnv) && !languageFromEnv.empty())
+    {
+        wxVector<wxString> preferred;
+        wxStringTokenizer tknzr(languageFromEnv, ":");
+        while (tknzr.HasMoreTokens())
+        {
+            const wxString tok = tknzr.GetNextToken();
+            if (const wxLanguageInfo* li = wxUILocale::FindLanguageInfo(tok))
+            {
+                preferred.push_back(li->CanonicalName);
+            }
```

Comment:
Very minor, but maybe add a trace/debug message saying that we ignore unknown `tok` in `WX_LANGUAGE`?

Reply:
I took this code from UNIX implementation, so it would make sense to add tracing there, but there's the logic more complicated and would significantly inflate the function... At that point I realized that the trace output lists detected system languages anyway, so any misparsing should be pretty obvious if you compare it to WXLANGUAGE value — which I changed the code to include in the trace message.